AutomatedTesting/Gem/PythonTests/Terrain/EditorScripts/Terrain_World_ConfigurationWorks.py

```diff
@@ -0,0 +1,163 @@
+
+    Test Steps:
+     1) Load the base level
+     2) Load the level components
+     3) Create 2 test entities, one parent at 512.0, 512.0, 50.0 and one child at the default position and add the required components
+     4) Start the Tracer to catch any errors and warnings
+     5) Set the base Terrain World values
+     6) Change the Axis Aligned Box Shape dimensions
+     7) Set the Vegetation Shape reference to terrain_spawner_entity
+     8) Set the FastNoise Gradient frequency to 0.01
+     9) Set the Gradient List to height_provider_entity
+     10) Disable and Enable the Terrain Gradient List so that it is recognised
+     11) Check terrain exists at a known position in the world
+     12) Check terrain does not exist at a known position outside the world
+     13) Check height value is the expected one when query resolution is changed
+    """
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import TestHelper as helper, Report
+    from editor_python_test_tools.utils import Report, Tracer
+    import editor_python_test_tools.hydra_editor_utils as hydra
+    import azlmbr.math as azmath
+    import azlmbr.legacy.general as general
+    import azlmbr.bus as bus
+    import azlmbr.editor as editor
+    import azlmbr.terrain as terrain
+    import math
+
+    SET_BOX_X_SIZE = 2048.0
+    SET_BOX_Y_SIZE = 2048.0
+    SET_BOX_Z_SIZE = 100.0
+    CLAMP = 1
+
+    helper.init_idle()
+    
+    # 1) Load the level
+    helper.open_level("", "Base")
+    helper.wait_for_condition(lambda: general.get_current_level_name() == "Base", 2.0)
+
+    # 2) Load the level components
+    terrain_world_component = hydra.add_level_component("Terrain World")
+    hydra.add_level_component("Terrain World Renderer")
+
+    # 3) Create 2 test entities, one parent at 512.0, 512.0, 50.0 and one child at the default position and add the required components
+    entity1_components_to_add = ["Axis Aligned Box Shape", "Terrain Layer Spawner", "Terrain Height Gradient List", "Terrain Physics Heightfield Collider"]
+    entity2_components_to_add = ["Vegetation Reference Shape", "Gradient Transform Modifier", "FastNoise Gradient"]
+    terrain_spawner_entity = hydra.Entity("TerrainEntity")
+    terrain_spawner_entity.create_entity(azmath.Vector3(512.0, 512.0, 50.0), entity1_components_to_add)
+    Report.result(Tests.create_terrain_spawner_entity, terrain_spawner_entity.id.IsValid())
+    height_provider_entity = hydra.Entity("HeightProviderEntity")
+    height_provider_entity.create_entity(azmath.Vector3(0.0, 0.0, 0.0), entity2_components_to_add,terrain_spawner_entity.id)
+    Report.result(Tests.create_height_provider_entity, height_provider_entity.id.IsValid())
+
+    # Give everything a chance to finish initializing.
+    general.idle_wait_frames(1)
+
+    # 4) Start the Tracer to catch any errors and warnings
+    with Tracer() as section_tracer:
```

Comment:
why are the entity and component setup not within tracer?

Reply:
No reason except that was the way the example I was looking at did it. Changed.